plugins/base/src/main/kotlin/translators/psi/SynheticElementDocumentationProvider.kt

```diff
@@ -0,0 +1,41 @@
+package org.jetbrains.dokka.base.translators.psi
+
+import com.intellij.psi.*
+import com.intellij.psi.javadoc.PsiDocComment
+import org.jetbrains.dokka.analysis.DokkaResolutionFacade
+import org.jetbrains.dokka.base.translators.psi.parsers.JavadocParser
+import org.jetbrains.dokka.model.doc.DocumentationNode
+
+private const val ENUM_VALUEOF_TEMPLATE_PATH = "/dokka/docs/javadoc/EnumValueOf.java.template"
+private const val ENUM_VALUES_TEMPLATE_PATH = "/dokka/docs/javadoc/EnumValues.java.template"
+
+internal class SyntheticElementDocumentationProvider(
+    private val javadocParser: JavadocParser,
+    private val resolutionFacade: DokkaResolutionFacade
+) {
+    fun isDocumented(psiElement: PsiElement): Boolean = psiElement is PsiMethod
+            && (psiElement.isSyntheticEnumValuesMethod() || psiElement.isSyntheticEnumValueOfMethod())
+
+    fun getDocumentation(psiElement: PsiElement): DocumentationNode? {
+        val psiMethod = psiElement as? PsiMethod ?: return null
+        val templatePath = when {
+            psiMethod.isSyntheticEnumValuesMethod() -> ENUM_VALUES_TEMPLATE_PATH
+            psiMethod.isSyntheticEnumValueOfMethod() -> ENUM_VALUEOF_TEMPLATE_PATH
+            else -> return null
+        }
+        val docComment = loadSyntheticDoc(psiElement, templatePath) ?: return null
+        return javadocParser.parseDocComment(docComment, psiElement)
+    }
+
+    private fun loadSyntheticDoc(psiElement: PsiElement, path: String): PsiDocComment? {
+        val containingClassName = (psiElement as? PsiMember)?.containingClass?.name ?: return null
+        val template = javaClass.getResource(path)?.readText() ?: return null
+        val text = template.replace("<ClassName>", containingClassName)
```

Comment:
Why does it need?

Reply:
Nice catch! Removed it.

I initially copy-pasted the template [from IntelliJ](https://github.com/JetBrains/intellij-community/blob/0af2e7b5c59e57c3a0080d736d5a6732a540407e/resources/src/javadoc/EnumValues.java.template#L6), and it had a usage example:
```kotlin
 * <pre>
 *    for(<ClassName> c : <ClassName>.values())
 *        System.out.println(c);
 * </pre>
```

But I removed the example since there's no distinction between Kotlin and Java.